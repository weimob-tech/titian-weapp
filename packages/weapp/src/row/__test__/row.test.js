const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', () => {
    const comp = getComponent('row');
    expect(comp).toMatchSnapshot();
  });

  it('should gutter prop render success', () => {
    const comp = getComponent('row', {
      gutter: 8
    });

    const row = comp.querySelector('.titian-row-flex').dom;

    expect(row.style.marginLeft).toBe('-4px');
    expect(row.style.marginRight).toBe('-4px');

    expect(comp).toMatchSnapshot();
  });

  it('should flex prop render success', () => {
    const comp = getComponent('row', {
      flex: false
    });

    expect(comp).toMatchSnapshot();
  });
});
