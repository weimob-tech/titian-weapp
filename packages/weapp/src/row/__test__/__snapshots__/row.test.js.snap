// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should flex prop render success 1`] = `
<main>
  <wx-view
    class="titian-row"
    style="margin-left: -0rpx; margin-right: -0rpx;"
  />
</main>
`;

exports[`test component should gutter prop render success 1`] = `
<main>
  <wx-view
    class="titian-row-flex titian-row"
    style="margin-left: -4rpx; margin-right: -4rpx;"
  />
</main>
`;

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="titian-row-flex titian-row"
    style="margin-left: -0rpx; margin-right: -0rpx;"
  />
</main>
`;
